spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/DefaultArgumentsNamespacingStrategy.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.launcher;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.boot.bind.PropertiesConfigurationFactory;
+import org.springframework.core.env.PropertySources;
+
+/**
+ * A simple implementation of {@link ArgumentsNamespacingStrategy} that simply prefixes each module
+ * argument with '{@literal <module key>.}'.
+ *
+ * <p>As an example, this is how one could specify the {@literal format} argument to the canonical time source:
+ * <pre>
+ *     org.springframework.cloud.stream.module:time-source:1.0.0.BUILD-SNAPSHOT.format=hh:mm:ss
```

Comment:
Seems what we'd want is something that would directly be the "label" from XD/s-c-d when used in that context.
What about turning modules into a `Map<label, mavenCoordinates>` and arguments into `Map<label, Map<String, String>>` ?